akka-http2-support/src/test/scala/akka/http/impl/engine/http2/WithPriorKnowledgeSpec.scala

```diff
@@ -45,47 +38,29 @@ class WithPriorKnowledgeSpec extends AkkaSpec("""
     "respond to cleartext HTTP/2 requests with cleartext HTTP/2" in {
       val (host, port) = (binding.localAddress.getHostName, binding.localAddress.getPort)
 
-      val (source, sink) =
+      val fromServer = Http2FrameProbe()
+
+      val source =
         Source.queue[String](1000, OverflowStrategy.fail)
           .map(str => ByteString(Base64.getDecoder.decode(str)))
           .via(Tcp().outgoingConnection(host, port))
-          .toMat(Sink.queue())(Keep.both)
+          .toMat(fromServer.sink)(Keep.left)
           .run()
 
       // Obtained by converting the input request bytes from curl with --http2-prior-knowledge
       // This includes port 9009 as 'authority', which our server accepts.
       source.offer("UFJJICogSFRUUC8yLjANCg0KU00NCg0KAAASBAAAAAAAAAMAAABkAARAAAAAAAIAAAAAAAAECAAAAAAAP/8AAQAAHgEFAAAAAYKEhkGKCJ1cC4Fw3HwAf3qIJbZQw6u20uBTAyovKg==").futureValue
 
-      // read settings frame
-      val responseSettings = sink.pull().futureValue
```

Comment:
That pulled probably in more than one frame (coalesced by TCP) so the next assertions were failing.